### DISCUSSION TITLE
Moving away from Travis CI

### DISCUSSION THREAD

**Discussion by igordejanovic** [textX/textX#347] [General]
I've been experiencing an enormous slowdown on our Travis CI tests in the last few months, to the point that it becomes useless (being queued for 5 hours just to run a set of tests that locally run in a few seconds is no fun).

Furthermore, [Travis CI is changing the pricing model](https://news.ycombinator.com/item?id=24964601) which makes their support for FLOSS projects questionable.

There are other alternatives which we could consider, for example [GitHub Actions](https://github.com/features/actions) or [Azure Pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/).

[An interesting blog post on the subject](https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works)

**Comment by goto40** [comment]
I heard about github actions in podcasts.

Since the codebase is already at github, why not also build there... 

But I have no real preference.

**Comment by stanislaw** [comment]
We have been using GitHub Actions for quite a while. I can recommend it.

1) https://github.com/strictdoc-project/strictdoc/tree/master/.github/workflows (Python)
2) https://github.com/mull-project/mull/tree/master/.github/workflows (C++, with Ansible for tasks)
